### ISSUE TITLE
Add intrinsic for pause instruction

### ISSUE THREAD

**Issue by Tetralux** [ziglang/zig#3162]
I believe it's called `llvm.x86.sse2.pause`.

Exerpt from https://www.felixcloutier.com/x86/pause:
> Improves the performance of spin-wait loops. When executing a “spin-wait loop,” processors will suffer a severe performance penalty when exiting the loop because it detects a possible memory order violation. The PAUSE instruction provides a hint to the processor that the code sequence is a spin-wait loop. The processor uses this hint to avoid the memory order violation in most situations, which greatly improves processor performance. For this reason, it is recommended that a PAUSE instruction be placed in all spin-wait loops.
>
> An additional function of the PAUSE instruction is to reduce the power consumed by a processor while executing a spin loop. A processor can execute a spin-wait loop extremely quickly, causing the processor to consume a lot of power while it waits for the resource it is spinning on to become available. Inserting a pause instruction in a spin-wait loop greatly reduces the processor’s power consumption.

Also: https://software.intel.com/sites/default/files/managed/39/c5/325462-sdm-vol-1-2abcd-3abcd.pdf#G13.31830

**Comment by CashWasabi** [comment]
Hey, my knowledge about this topic is kind of limited. Is this about `std.atomic.spinLoopHint()` and is this not supported yet? I'm trying to work have some threads in a hot state while working through data and was wondering if there still is a performance penalty without this pause instruction. If this is already solved or not related at all just ignore my comment/question.

**Comment by leecannon** [comment]
I think this issue should be closed, as shown below `spinLoopHint()` already uses the pause instruction and matching instructions on other architectures. There is no need for a builtin.
https://github.com/ziglang/zig/blob/2962db333f43c8bb10a1e2ad4cdd19dfab26515b/lib/std/atomic.zig#L348-L355

**Comment by alexrp** [comment]
Agreed, I don't think having a builtin adds anything over using inline assembly.
